services/youtube/youtube-base.js

```diff
@@ -0,0 +1,71 @@
+'use strict'
+
+const Joi = require('@hapi/joi')
+const { BaseJsonService, NotFound } = require('..')
+const { metric } = require('../text-formatters')
+const { nonNegativeInteger } = require('../validators')
+
+const schema = Joi.object({
+  items: Joi.array()
+    .items(
+      Joi.object({
+        statistics: Joi.object({
+          viewCount: nonNegativeInteger,
+          likeCount: nonNegativeInteger,
+          dislikeCount: nonNegativeInteger,
+          commentCount: nonNegativeInteger,
+        }).required(),
+      })
+    )
+    .required(),
+}).required()
+
+module.exports = class YouTubeBase extends BaseJsonService {
+  static get category() {
+    return 'other'
+  }
+
+  static get auth() {
+    return {
+      passKey: 'youtube_api_key',
+      authorizedOrigins: ['https://www.googleapis.com'],
+      isRequired: true,
+    }
+  }
+
+  static get defaultBadgeData() {
+    return { label: 'youtube', color: 'red' }
```

Comment:
Thoughts on making this a "social" styled badge, like the Twitter one?

Reply:
That sounds reasonable to me, especially since we're embedding the link (which AFAIK is something we only do on social badges)

Reply:
Done!